src/SprykerEco/Zed/AvalaraTax/Business/AvalaraTaxBusinessFactory.php

```diff
@@ -0,0 +1,214 @@
+    {
+        return new SingleShipmentCartItemAvalaraTaxCalculator(
+            $this->createAvalaraTransactionExecutor(),
+            $this->getMoneyFacade(),
+            $this->getUtilEncodingService()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Calculator\CartItemAvalaraTaxCalculatorInterface
+     */
+    public function createMultiShipmentCartItemTaxCalculator(): CartItemAvalaraTaxCalculatorInterface
+    {
+        return new MultiShipmentCartItemAvalaraTaxCalculator(
+            $this->createAvalaraTransactionExecutor(),
+            $this->getMoneyFacade(),
+            $this->getUtilEncodingService()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Executor\AvalaraTransactionExecutorInterface
+     */
+    public function createAvalaraTransactionExecutor(): AvalaraTransactionExecutorInterface
+    {
+        return new AvalaraTransactionExecutor(
+            $this->createAvalaraTransactionBuilder(),
+            $this->createAvalaraTransactionRequestMapper(),
+            $this->createAvalaraTransactionResponseMapper(),
+            $this->createAvalaraTransactionLogger()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Mapper\AvalaraTransactionRequestMapperInterface
+     */
+    public function createAvalaraTransactionRequestMapper(): AvalaraTransactionRequestMapperInterface
+    {
+        return new AvalaraTransactionRequestMapper(
+            $this->getConfig(),
+            $this->getMoneyFacade(),
+            $this->getCreateTransactionRequestExpanderPlugins()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Mapper\AvalaraTransactionResponseMapperInterface
+     */
+    public function createAvalaraTransactionResponseMapper(): AvalaraTransactionResponseMapperInterface
+    {
+        return new AvalaraTransactionResponseMapper($this->getUtilEncodingService());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Builder\AvalaraTransactionBuilderInterface
+     */
+    public function createAvalaraTransactionBuilder(): AvalaraTransactionBuilderInterface
+    {
+        return new AvalaraTransactionBuilder($this->createTransactionBuilder());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Logger\AvalaraTransactionLoggerInterface
+     */
+    public function createAvalaraTransactionLogger(): AvalaraTransactionLoggerInterface
+    {
+        return new AvalaraTransactionLogger(
+            $this->getEntityManager(),
+            $this->getStoreFacade(),
+            $this->getUtilEncodingService()
+        );
+    }
+
+    /**
+     * @return \Avalara\TransactionBuilder
+     */
+    public function createTransactionBuilder(): TransactionBuilder
+    {
+        return new TransactionBuilder(
+            $this->createTransactionClient(),
+            $this->getConfig()->getCompanyCode(),
+            (string)DocumentType::C_SALESORDER,
```

Comment:
* It's better to extract this constant to the module config with reference to the original one.

Reply:
Resolved.